lib/dependencies/CommonJsRequireDependencyParserPlugin.js

```diff
@@ -73,53 +73,58 @@ class CommonJsRequireDependencyParserPlugin {
 				parser.state.current.addDependency(dep);
 				return true;
 			});
-		parser.hooks.call
-			.for("require")
-			.tap("CommonJsRequireDependencyParserPlugin", expr => {
-				if (expr.arguments.length !== 1) return;
-				let localModule;
-				const param = parser.evaluateExpression(expr.arguments[0]);
-				if (param.isConditional()) {
-					let isExpression = false;
-					const prevLength = parser.state.current.dependencies.length;
-					const dep = new RequireHeaderDependency(expr.callee.range);
-					dep.loc = expr.loc;
-					parser.state.current.addDependency(dep);
-					for (const p of param.options) {
-						const result = processItem(expr, p);
-						if (result === undefined) {
-							isExpression = true;
-						}
-					}
-					if (isExpression) {
-						parser.state.current.dependencies.length = prevLength;
-					} else {
-						return true;
+
+		const handler = expr => {
+			if (expr.arguments.length !== 1) return;
+			let localModule;
+			const param = parser.evaluateExpression(expr.arguments[0]);
+			if (param.isConditional()) {
+				let isExpression = false;
+				const prevLength = parser.state.current.dependencies.length;
+				const dep = new RequireHeaderDependency(expr.callee.range);
+				dep.loc = expr.loc;
+				parser.state.current.addDependency(dep);
+				for (const p of param.options) {
+					const result = processItem(expr, p);
+					if (result === undefined) {
+						isExpression = true;
 					}
 				}
-				if (
-					param.isString() &&
-					(localModule = LocalModulesHelpers.getLocalModule(
-						parser.state,
-						param.string
-					))
-				) {
-					const dep = new LocalModuleDependency(localModule, expr.range);
-					dep.loc = expr.loc;
-					parser.state.current.addDependency(dep);
-					return true;
+				if (isExpression) {
+					parser.state.current.dependencies.length = prevLength;
 				} else {
-					const result = processItem(expr, param);
-					if (result === undefined) {
-						processContext(expr, param);
-					} else {
-						const dep = new RequireHeaderDependency(expr.callee.range);
-						dep.loc = expr.loc;
-						parser.state.current.addDependency(dep);
-					}
 					return true;
 				}
-			});
+			}
+			if (
+				param.isString() &&
+				(localModule = LocalModulesHelpers.getLocalModule(
+					parser.state,
+					param.string
+				))
+			) {
+				const dep = new LocalModuleDependency(localModule, expr.range);
```

Comment:
~You probably need to use `[expr.callee.range[0], expr.range[1]]` as range instead to exclude the `new`~

This wouldn't work for `new (require)("local-module")`.

Better add a `callNew` property to the `LocalModuleDependency` and modify the template to add a `new` infront.

Reply:
Ok, will have to dive deeper into it then.